### PR TITLE
[POT] New IE API integration

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Details:
 - *Replaced IE imports with the openvino.runtime*
 - *Added sync and async execution for the model with the new API*

### Tickets:
 - *68382*

### CODE REVIEW

File: tools/pot/openvino/tools/pot/benchmark/infer_request_wrap.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def callback(self, statusCode, userdata):
         self.callback_queue(self.id, self.request.latency)
 
     def start_async(self, input_data):
-        self.request.async_infer(input_data)
+        self.request.start_async(input_data)
 
     def infer(self, input_data):
```
Comment (ilya-lavrenov) [inline tools/pot/openvino/tools/pot/benchmark/infer_request_wrap.py:27]: Can the async infer queue be reused from python api?

@jiwaszki

Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/benchmark/infer_request_wrap.py:27]: Is this already introduced, validated, and used by other components? I heard that this feature is delayed that is why we decided to go via legacy way. But we will move to it in the future.

Comment (jiwaszki) [reply tools/pot/openvino/tools/pot/benchmark/infer_request_wrap.py:27]: @ilya-lavrenov @AlexKoff88 Actually it is introduced in https://github.com/openvinotoolkit/openvino/pull/8417 and I am awaiting reviews. It is already used by benchmark_app (pure py-implementation).

File: tools/pot/openvino/tools/pot/graph/graph_utils.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def load_graph(model_config, target_device='ANY'):
     xml_path = model_config.model
 
     if target_device in special_transform_devices:
-        network = ie.read_network(model=xml_path, weights=bin_path)
+        network = ie.read_model(model=xml_path, weights=bin_path)
         ApplyPOTTransformations(network, target_device.encode('utf-8'))
```
Comment (ilya-lavrenov) [inline tools/pot/openvino/tools/pot/graph/graph_utils.py:32]: let's use new offline API module like here https://github.com/openvinotoolkit/openvino/pull/9121/files

File: tools/pot/openvino/tools/pot/graph/graph_utils.py [hunk 3]
```diff
@@ -28,11 +28,11 @@ def load_graph(model_config, target_device='ANY'):
     xml_path = model_config.model
 
     if target_device in special_transform_devices:
-        network = ie.read_network(model=xml_path, weights=bin_path)
-        ApplyPOTTransformations(network, target_device.encode('utf-8'))
+        network = ie.read_model(model=xml_path, weights=bin_path)
```
Comment (ilya-lavrenov) [inline tools/pot/openvino/tools/pot/graph/graph_utils.py:31]: @GlebKazantaev should we use FE API directly here?

File: tools/pot/openvino/tools/pot/engines/ie_engine.py [hunk 4]
```diff
@@ -323,10 +331,13 @@ def _process_dataset_async(self, stats_layout, sampler, print_progress=False,
         progress_log_fn = logger.info if print_progress else logger.debug
 
         # Load model to the plugin
-        executable_model = self._ie.load_network(network=self._model,
-                                                 device_name=self.config.device,
-                                                 num_requests=requests_num)
-        free_irs = executable_model.requests
+        executable_model = self._ie.compile_model(model=self._model, device_name=self.config.device)
```
Comment (AlexKoff88) [inline tools/pot/openvino/tools/pot/engines/ie_engine.py:334]: Now it looks like this is a compiled model. The object type is also called CompiledModel

File: tools/pot/openvino/tools/pot/engines/ac_engine.py [hunk 5]
```diff
@@ -123,14 +125,31 @@ def predict(self, stats_layout=None, sampler=None, stat_aliases=None,
...
-            model_output_names = [out for m_dict in self._model for out in m_dict['model'].outputs.keys()]
+            model_output_names = []
+            model_outputs = []
+            for model in self._model:
+                model_outputs.extend(model['model'].outputs)
+            for ng_output in model_outputs:
+                model_output_names.extend(list(ng_output.get_tensor().get_names()))
+
+            nodes_name = []
+            for names_map in nodes_names_map.values():
```
Comment (AlexKoff88) [inline tools/pot/openvino/tools/pot/engines/ac_engine.py:150]: I noticed some of the code that is responsible for tensor names, etc. is duplicated in both Engines (AC and IE). Can we extract it in a meaningful method and reuse?

Comment (KodiaqQ) [reply tools/pot/openvino/tools/pot/engines/ac_engine.py:150]: Yes, working on it. Thanks for the notice.

**Review by l-bat — APPROVED** [review]

**Review by MilanaShhanukova — APPROVED** [review]

File: tools/pot/openvino/tools/pot/algorithms/quantization/bias_correction/algorithm.py [hunk 6]
```diff
@@ -283,8 +283,11 @@ def _create_results_after_nodes(self, output_nodes):
         outputs_data = []
         for output_node in output_nodes:
             output_name = nu.create_node_name(output_node, str)
-            result_name = output_name + '/result'
+            result_name = output_name + '/sink_port_0'
```
Comment (AlexKoff88) [inline tools/pot/openvino/tools/pot/algorithms/quantization/bias_correction/algorithm.py:286]: @KodiaqQ, are we breaking compatibility with IRv10 with this change? Can you take a look at the changes and highlight what can break it?

**Pull request merged**